File: prompts/fixtures/python/debug_test.py
import sys


def add(a, b):
    result = a + b        # line 5
    return result          # line 6


def multiply(a, b):
    result = a * b        # line 10
    return result          # line 11


def compute(x, y):
    sum_ = add(x, y)              # line 15
    product = multiply(x, y)      # line 16
    final = sum_ + product        # line 17
    return final                   # line 18


def loop_sum(n):
    total = 0                      # line 22
    for i in range(1, n + 1):
        total = add(total, i)     # line 24
    return total                   # line 25


def factorial(n):
    if n <= 1:                     # line 29
        return 1                   # line 30
    return n * factorial(n - 1)   # line 31


def main():
    x = 10                                 # line 35
    y = 20                                 # line 36
    result1 = compute(x, y)               # line 37
    print(f"compute = {result1}")          # line 38
    result2 = loop_sum(5)                  # line 39
    print(f"loop_sum = {result2}")         # line 40
    result3 = factorial(5)                 # line 41
    print(f"fact = {result3}")             # line 42
    return 0                               # line 43


if __name__ == "__main__":
    sys.exit(main())

File: prompts/fixtures/rust/debug_test.rs
fn add(a: i32, b: i32) -> i32 {
    let result = a + b;        // line 2
    result                      // line 3
}

fn multiply(a: i32, b: i32) -> i32 {
    let result = a * b;        // line 7
    result                      // line 8
}

fn compute(x: i32, y: i32) -> i32 {
    let sum = add(x, y);              // line 12
    let product = multiply(x, y);     // line 13
    let final_ = sum + product;       // line 14
    final_                             // line 15
}

fn loop_sum(n: i32) -> i32 {
    let mut total = 0;                 // line 19
    for i in 1..=n {
        total = add(total, i);        // line 21
    }
    total                              // line 23
}

fn factorial(n: i32) -> i32 {
    if n <= 1 {                        // line 27
        return 1;                      // line 28
    }
    n * factorial(n - 1)              // line 30
}

fn main() {
    let x = 10;                                    // line 34
    let y = 20;                                    // line 35
    let result1 = compute(x, y);                  // line 36
    println!("compute = {}", result1);            // line 37
    let result2 = loop_sum(5);                     // line 38
    println!("loop_sum = {}", result2);           // line 39
    let result3 = factorial(5);                    // line 40
    println!("fact = {}", result3);               // line 41
}
